subjects/java/checkpoints/date-formatter/README.md

```diff
@@ -0,0 +1,72 @@
+## Date Formatter
+
+### Instructions
+
+Write a class `DateFormatter` that has three private attributes:
+
+- `date`: the date to be formatted (as a date).
+- `format`: the format to convert the date to.
+- `formattedDate`: contains the date converted into the given format.
+
+Create getters for all the attributes and setters for `date` and `format`. The conversion should happen when you set the `format` or the `date`. If the `format` is not correct, the conversion shouldn't happen. Additionally, create methods to output the date in the specified `format`, and constructors with only the `date`, with `date` and `format`, and a default constructor.
+
+#### The accepted date formats are:
+
+- `DD/MM/YYYY`
+- `DD Month YYYY`
+- `DD.MM.YYYY`
+
+#### The accepted date formats are:
```

Comment:
What should happen if the format is not correct?
Should the constructor (or setter) of `format` throw an error or the method `DateFormatter.getFormattedDate` should take care of the error handling?

What is the expected error, log ... ?

Reply:
this is specified in the subject in case the format is not correct the conversion will not applied so the `formattedDate` attribute will keep its old value.